### PR TITLE
Add linter warning for simple domains used in loops

### PR DESCRIPTION
**Pull request by jabraham17**
Adds a linter warning for simple domains used in loops. Domain literals like `{1..10}` used as the iterand of a loop cause an unnecessary domain creation. Although this may be optimized away by the compiler, it is cleaner and possibly more performant to not have the domain literal.

Screenshot 2024-04-04 at 9 31 08 AM

[Reviewed by @DanilaFe]

### CODE REVIEW

File: tools/chplcheck/src/rules.py [hunk 1]
```diff
@@ -323,3 +323,43 @@ def variables(node):
...
+            if (
+                isinstance(node, OpCall)
+                and node.is_binary_op()
+                and (
+                    node.op() == "#"
+                    or node.op() == "by"
+                    or node.op() == "align"
+                )
+            ):
+                return is_range_like(list(node.actuals())[0])
```
Comment (DanilaFe) [inline tools/chplcheck/src/rules.py:351]: Can we just add an `actual(idx)` getter... This is really unfortunate.
Alternatively, `next(node.actuals())` should work.

File: tools/chplcheck/src/rules.py [hunk 2]
```diff
@@ -323,3 +323,43 @@ def variables(node):
...
+            if not isinstance(iterand, Domain):
+                continue
+            exprs = list(iterand.exprs())
+            if len(exprs) != 1:
+                continue
+            # only warn for ranges or count operators
+            if not is_range_like(exprs[0]):
+                continue
+
+            yield iterand
```
Comment (DanilaFe) [inline tools/chplcheck/src/rules.py:365]: Consider yielding an anchor here to allow silencing.

**Review by DanilaFe — APPROVED** [review]
Nice, just a couple of minor comments.

File: tools/chplcheck/src/rules.py (hunk 1 above)
Comment (jabraham17) [reply tools/chplcheck/src/rules.py:351]: I think its perfectly reasonable to add a `actual(idx)` getter. I'll add to this PR

**Review by DanilaFe — APPROVED** [review]
Thanks!

**Pull request merged**